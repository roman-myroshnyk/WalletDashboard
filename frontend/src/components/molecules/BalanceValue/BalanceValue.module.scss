@import '@/styles/colors.scss';

.balanceValue {
  color: $gray;
  font-weight: bold;
  font-size: 22px;
  font-family: Epilogue, sans-serif;
  font-style: normal;
  line-height: 22px;
  text-decoration: none;
}
