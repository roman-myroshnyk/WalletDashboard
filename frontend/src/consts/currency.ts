export enum CurrencyLables {
    USD = 'USD',
    EUR = 'EUR',
  }

export enum DigitalCurrencyLabels {
  ETH = 'ETH',
  BTC = 'BTC'
}
export enum CurrencySymbols {
    USD = '$',
    EUR = 'â‚¬',
  }

export type CurrencyInfo = {
    symbol: CurrencySymbols,
    label: CurrencyLables,
    description: string
  }
export const EnabledCurrencies:CurrencyInfo[] = [
  {
    symbol: CurrencySymbols.USD,
    label: CurrencyLables.USD,
    description: 'United States Dollar',
  },
  {
    symbol: CurrencySymbols.EUR,
    label: CurrencyLables.EUR,
    description: 'Euro Member Countries',
  },
];

export const CurrencyNames = EnabledCurrencies.map((currency) => currency.label);

type TCurrencyIndexes = { [K in keyof typeof CurrencyLables]: number }

export const CurrencyIndexes: TCurrencyIndexes = EnabledCurrencies.reduce(
  (indexes, currencyInfo, currentIndex) => (
    { ...indexes, [currencyInfo.label]: currentIndex }),
  { } as TCurrencyIndexes,
);
