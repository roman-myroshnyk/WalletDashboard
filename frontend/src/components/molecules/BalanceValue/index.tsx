import { FC } from 'react';
import { useSpring, animated, config } from 'react-spring';
import { CurrencySymbols } from '@/consts/currency';
import styles from './BalanceValue.module.scss';

export interface IProps {
    value: number | null;
    currencySymbol?:CurrencySymbols;
    testid?: string;
}

const BalanceValue:FC<IProps> = ({ value, currencySymbol, testid }) => {
  const { animatedValue } = useSpring({ animatedValue: value || 0, config: config.molasses });
  return (
    <strong className={styles.balanceValue} data-testid={testid}>
      <animated.span>
        {animatedValue.to((val) => (val.toFixed(2)))}
      </animated.span>
      {` ${currencySymbol || ''}`}
    </strong>
  );
};

BalanceValue.defaultProps = {
  currencySymbol: CurrencySymbols.USD,
  testid: 'balanceValue',
};
export default BalanceValue;
