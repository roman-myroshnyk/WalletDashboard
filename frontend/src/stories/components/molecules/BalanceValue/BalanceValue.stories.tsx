import { Meta, Story } from '@storybook/react/types-6-0';
import BalanceValue, { IProps } from '@/molecules/BalanceValue';
import { CurrencySymbols } from '@/consts/currency';

export default {
  title: 'molecules/BalanceValue',
  component: BalanceValue,
  argTypes: {
    value: {
      control: 'number',
    },
    currencySymbol: {
      options: CurrencySymbols,
      control: 'select',
    },
  },
} as Meta;

const Template:Story<IProps> = (args) => (
  <BalanceValue {...args} />
);

export const Default = Template.bind({});

Default.args = {
  value: 2,
  currencySymbol: CurrencySymbols.EUR,
};
