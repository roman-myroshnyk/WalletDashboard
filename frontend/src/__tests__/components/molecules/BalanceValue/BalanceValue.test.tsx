import '.../../../mocks/framer-motion';
import { render } from '@testing-library/react';
import BalanceValue from '@/molecules/BalanceValue';
import { CurrencySymbols } from '@/consts/currency';

describe('molecules/BalanceValue', () => {
  it('BalanceValue should render', () => {
    const { container } = render(
      <BalanceValue
        value={14}
        currencySymbol={CurrencySymbols.EUR}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
