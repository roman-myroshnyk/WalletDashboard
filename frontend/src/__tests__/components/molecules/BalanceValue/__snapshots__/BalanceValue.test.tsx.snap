// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/BalanceValue BalanceValue should render 1`] = `
<div>
  <strong
    class="balanceValue"
  >
    <span>
      14
    </span>
     â‚¬
  </strong>
</div>
`;
